docs/multi_agents.md

```diff
@@ -0,0 +1,52 @@
+# What are AI Agents?
+AI agents are autonomous software systems that can perceive their environment, make decisions, and take actions to achieve specific goals. They are designed to operate without human intervention, leveraging advanced AI techniques like natural language processing and machine learning.
+## Key characteristics of AI agents
+* **Autonomy**: AI agents can make decisions and act independently based on their goals and the information they gather.
+* **Flexibility**: They can adapt to changing circumstances and learn from experience to improve their performance over time.
+* **Reactivity**: AI agents can perceive their environment and respond to changes in real-time.
+
+## Components of an AI Agent
+AI agents are composed of several key components that enable their autonomous behavior:
+Sensors: Used to gather information from the environment, such as user inputs, sensor data, or external APIs.
+Agent Function: The core decision-making algorithm that maps sensor inputs to actions.
+Actuators: Mechanisms that allow the agent to affect its environment, such as generating text responses, making API calls, or sending notifications.
+Goals: The objectives the agent is designed to achieve, which guide its decision-making process.
+
+## What is a multi-agent AI framework?
+A multi-agent AI framework is a system where multiple intelligent agents interact or work together to perform complex tasks, solve problems, or simulate environments. These agents can be autonomous entities, each with its own capabilities, knowledge base, and goals. Here’s a detailed description of a multi-agent AI framework:
+
+## Key Components of a multi-agent AI framework
+* **Agents**:
+    * Reactive Agents: Respond to changes in the environment.
+    * Deliberative Agents: Use reasoning and planning to achieve goals.
+    * Hybrid Agents: Combine reactive and deliberative strategies.
+* **Communication Mechanism**:
+    * Direct Communication: Agents send messages to each other.
+    * Indirect Communication: Agents use shared context or memory.
+* **Coordination and Cooperation**:
+Strategies for agents to work together towards common goals, avoid conflicts, and optimize joint performance.
+    * Examples:
+        * Task Allocation: Dividing tasks among agents based on their capabilities.
+        * Task planning and prioritization
+        * Negotiation/reasoning: Agents discuss to reach agreements.
+        * Conflict resolution: Avoid loop or infinite loop for the same question
+        * Challenge other agents: challenge previous steps to reduce hallucinations
+        * Decision-Making: Methods for agents to make choices based on their goals, perceptions, and available information. LLMs or other AI algorithms (tule engine, game theory etc) can be used.
+* **Retrieve additional information**: RAG agents can retrieve informations for better results. The agentic approach permit also a multi-step RAG refining progressively the information retrieved 
```

Comment:
s/informations/information

And, "The agentic approach also permits a multi-step..." sounds better :-)